modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/loader/parser/java/JavaAttributesResolverModelParser.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.module.extension.internal.loader.parser.java;
+
+import static org.mule.runtime.api.i18n.I18nMessageFactory.createStaticMessage;
+import static org.mule.runtime.core.api.util.ClassUtils.getClassName;
+
+import org.mule.runtime.api.exception.MuleRuntimeException;
+import org.mule.runtime.core.api.util.ClassUtils;
+import org.mule.runtime.module.extension.internal.loader.parser.AttributesResolverModelParser;
+import org.mule.runtime.module.extension.internal.metadata.SdkAttributesTypeResolverAdapter;
+import org.mule.sdk.api.metadata.NullMetadataResolver;
+import org.mule.sdk.api.metadata.resolving.AttributesTypeResolver;
+
+public class JavaAttributesResolverModelParser implements AttributesResolverModelParser {
+
+  private final Class<?> attributesTypeResolverDeclarationClass;
+  private final boolean muleResolver;
+
+  public JavaAttributesResolverModelParser(Class<?> attributesTypeResolverDeclarationClass, boolean muleResolver) {
```

Comment:
Ok, I have not seen that code. I may be wrong, but when moving the code to the parsers and remove the enricher, I think that the parsing of the annotations will have to be done before creating the actual java parser (which is the purpose of the parser). We can leave this as is, and it will be part of the refactoring issue to resolve this.

Reply:
You are right, and that is what I'm doing here. The `attributesTypeResolverDeclarationClass` is the class of the resolver.
The alternative is to have a new class, let called  Info, that holds the class of the resolver and a boolean that indicates what kind of resolver it is. This Info class can be created by the parser, in this way I can remove the `JavaOutputResolverModelParserUtils`
I think we are currently using both patters. For example we have `JavaExtensionModelParserUtils#getFunctionModelParsers` and `JavaExtensionModelParserUtils#getRequiresEnterpriseLicenseInfo`

Reply:
Great! So the code to create those java model parsers is within another java model parsers. My bad then, this is not an issue.